src/m365/entra/commands/m365group/m365group-set.spec.ts

```diff
@@ -381,6 +392,58 @@ describe(commands.M365GROUP_SET, () => {
     assert(loggerLogToStderrSpy.called);
   });
 
+  it('sets option allowExternalSenders when using delegated permissions', async () => {
+    const patchStub = sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { id: groupId, allowExternalSenders: true } });
+    assert(patchStub.firstCall.args[0].data.allowExternalSenders);
```

Comment:
Ideally, we should check for the entire body. Maybe to ignore the undefined properties we could do something like:

`const data = JSON.parse(JSON.stringify(postRequestBody))`

Let's fix this for other tests as well.